### PR TITLE
[grafana] - HA Alerting: Allow StatefulSet to be used instead of Deployment

### PR DESCRIPTION
**Pull request by jdziat**
For HA Alerting:
Would like the ability to use a StatefulSet instead of a deployment. That way I can dynamically populate the ha-peers using predetermined host names.

Example:
```
ha_peers=grafana-0:9094,grafana-1:9094,grafana-2:9094
```

### CODE REVIEW

**Comment by zanhsieh** [comment]
@jdziat 
This PR needs a version bump in Chart.yaml. Thank you in advance.

**Comment by jdziat** [comment]
@zanhsieh bumped

**Comment by zanhsieh** [comment]
@jdziat 
Please fix the CI error:
```
==> Linting charts/grafana
Error:  templates/: parse error at (grafana/templates/statefulset.yaml:1): unclosed left paren
```

File: charts/grafana/templates/statefulset.yaml [hunk 1]
```diff
@@ -1,4 +1,4 @@
-{{- if and .Values.persistence.enabled (not .Values.persistence.existingClaim) (eq .Values.persistence.type "statefulset")}}
+{{- if (or (.Values.useStatefulSet) (and .Values.persistence.enabled (not .Values.persistence.existingClaim) (eq .Values.persistence.type "statefulset"))}}
```
Comment (zanhsieh) [inline charts/grafana/templates/statefulset.yaml:1]:
```
{{- if (or (.Values.useStatefulSet) (and .Values.persistence.enabled (not .Values.persistence.existingClaim) (eq .Values.persistence.type "statefulset")))}}
```

**Review by zanhsieh — CHANGES_REQUESTED** [review]

**Review by zanhsieh — APPROVED** [review]

**Pull request merged**